src/main/java/me/roinujnosde/titansbattle/managers/DatabaseManager.java

```diff
@@ -66,59 +82,142 @@ private enum WinnerType {
     }
 
     public void setup() {
-        try {
-            Statement statement = getConnection().createStatement();
-            statement.execute("CREATE TABLE IF NOT EXISTS tb_warriors "
-                    + "(displayname varchar(30) NOT NULL,"
-                    + " uuid varchar(255) NOT NULL,"
-                    + " kills int NOT NULL,"
-                    + " deaths int NOT NULL,"
-                    + " victories int NOT NULL,"
-                    + " game varchar(20) NOT NULL);");
-            statement.execute("CREATE TABLE IF NOT EXISTS tb_groups"
-                    + "(identification varchar(255) NOT NULL,"
-                    + " kills int NOT NULL,"
-                    + " deaths int NOT NULL,"
-                    + " victories int NOT NULL,"
-                    + " defeats int NOT NULL,"
-                    + " game varchar(20) NOT NULL);");
-            statement.execute("CREATE TABLE IF NOT EXISTS tb_winners"
-                    + "(date varchar(10) NOT NULL,"
-                    + " killer varchar(255),"
-                    + " player_winners text,"
-                    + " winner_group varchar(255),"
-                    + " game varchar(20) NOT NULL);");
+        try (Statement stmt = getConnection().createStatement()) {
+            stmt.execute("CREATE TABLE IF NOT EXISTS tb_warriors "
+                              + "(displayname varchar(30) NOT NULL,"
+                              + " uuid varchar(255) NOT NULL,"
+                              + " kills int NOT NULL,"
+                              + " deaths int NOT NULL,"
+                              + " victories int NOT NULL,"
+                              + " game varchar(20) NOT NULL,"
+                              + " PRIMARY KEY (uuid, game));");
+            addPrimaryKeyIfNotExists("tb_warriors", "uuid, game");
+
+            stmt.execute("CREATE TABLE IF NOT EXISTS tb_groups"
+                              + "(identification varchar(255) NOT NULL,"
+                              + " kills int NOT NULL,"
+                              + " deaths int NOT NULL,"
+                              + " victories int NOT NULL,"
+                              + " defeats int NOT NULL,"
+                              + " game varchar(20) NOT NULL,"
+                              + " PRIMARY KEY (identification, game));");
+            addPrimaryKeyIfNotExists("tb_groups", "identification, game");
+
+            stmt.execute("CREATE TABLE IF NOT EXISTS tb_winners"
+                              + "(date varchar(10) NOT NULL,"
+                              + " killer varchar(255),"
+                              + " player_winners text,"
+                              + " winner_group varchar(255),"
+                              + " game varchar(20) NOT NULL,"
+                              + " PRIMARY KEY (date, game));");
+            addPrimaryKeyIfNotExists("tb_winners", "date, game");
```

Comment:
`date` and `game` are not unique.

Reply:
https://github.com/RoinujNosde/TitansBattle/pull/115#discussion_r1306688248